src/radical/entk/task.py

```diff
@@ -1,15 +1,17 @@
-
 __copyright__ = 'Copyright 2014-2020, http://radical.rutgers.edu'
 __license__   = 'MIT'
 
+import warnings
+
 import radical.utils as ru
 
 from . import exceptions as ree
 from . import states     as res
 
-import warnings
 warnings.simplefilter(action="once", category=DeprecationWarning, lineno=707)
 warnings.simplefilter(action="once", category=DeprecationWarning, lineno=764)
+warnings.simplefilter(action="once", category=DeprecationWarning, lineno=943)
```

Comment:
You added some lines here and above - doesn't that move the `lineno` entries above?  It's neat that this `once` thing exists, but is there a way to have this independent of line numbering?  The numbers at least don't point to the warning directives right now...

From https://docs.python.org/3/library/warnings.html#the-warnings-filter it seems that the `default` warning filter would also do the trick: print only once per module __and line__, w/o the need for explicit line numbers.

Reply:
I added the `once` because of #523. Before #523 I did not use `simplefilter` as I assumed that the `default` is what happens by default. :)

I will fix the linenumbers though.

Reply:
```py
#!/usr/bin/env python3

import warnings


warnings.simplefilter(action="default", category=DeprecationWarning)


def test():
    warnings.warn("warning 1", DeprecationWarning)


for _ in range(10):
    test()
```

This prints the warning once.

Reply:
I updated it locally and this is how task.py begins:
```python
"miniconda3/envs/rct_devel/lib/python3.7/site-packages/radical/entk/task.py" 1101L, 34001C       1,1           Top
   1 __copyright__ = 'Copyright 2014-2020, http://radical.rutgers.edu'
   2 __license__   = 'MIT'
   3
   4 import warnings
   5
   6 import radical.utils as ru
   7
   8 from string     import punctuation
   9 from .constants import NAME_MESSAGE
  10
  11 from . import exceptions as ree
  12 from . import states     as res
  13
  14 warnings.simplefilter(action="default", category=DeprecationWarning)
  15 #warnings.simplefilter(action="once", category=DeprecationWarning, lineno=724)
  16 #warnings.simplefilter(action="once", category=DeprecationWarning, lineno=782)
  17 #warnings.simplefilter(action="once", category=DeprecationWarning, lineno=950)
  18
  19
  20 # ------------------------------------------------------------------------------
```

After running the lfs example locally this is how the output looks like:
```
(rct_devel) iparask@js-17-185:~/EnTKUserGuide$ python lfs_tagging_dd.py
EnTK session: re.session.js-17-185.jetstream-cloud.org.iparask.018641.0000
Creating AppManagerSetting up RabbitMQ system                                 ok
Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
                                                                              ok
Validating and assigning resource manager                                     ok
CPU requirements keys are renamed. Please use cpu_processes for processes, cpu_process_type for process_type, cpu_threads for threads_per_process and cpu_thread_type for thread_type
CPU requirements keys are renamed. Please use cpu_processes for processes, cpu_process_type for process_type, cpu_threads for threads_per_process and cpu_thread_type for thread_type
Attribute tag is depcrecated. Use tags instead
CPU requirements keys are renamed. Please use cpu_processes for processes, cpu_process_type for process_type, cpu_threads for threads_per_process and cpu_thread_type for thread_type
Setting up RabbitMQ system                                                   n/a
new session: [re.session.js-17-185.jetstream-cloud.org.iparask.018641.0000]    \
database   : [mongodb://iparask:****@129.114.17.185:27017/iparask/]count is deprecated. Use estimated_document_count or count_documents instead. Please note that $where must be replaced by $expr, $near must be replaced
by $geoWithin with $center, and $nearSphere must be replaced by $geoWithin with $centerSphere
insert is deprecated. Use insert_one or insert_many instead.
update is deprecated. Use replace_one, update_one or update_many instead.
           ok
create pilot managerupdate is deprecated. Use replace_one, update_one or update_many instead.
insert is deprecated. Use insert_one or insert_many instead.
                                                          ok
submit 1 pilot(s)
        pilot.0000   local.localhost_anaconda      2 cores       0 gpusinitialize_ordered_bulk_op is deprecated
update is deprecated. Use replace_one, update_one or update_many instead.
```

This prints 3 warning from EnTK and several from other systems. This is under a tmux session called entk-test in our VM if you want to check.

@andre-merzky, @mturilli let me know how you want to proceed with printing deprecation messages?

Reply:
Oh well, if it does not work, it does not work :-/  Never mind my comment then...

Reply:
:D okay!